File: tests/fixtures/parity/emit.mjs
/**
 * Emisor de paridad — runtime JS.
 *
 * Resuelve el fixture canónico e imprime en stdout el JSON de valores,
 * ordenado por clave y con indentación de 2. Nada más va a stdout.
 *
 * Solo usa API pública común a ambos runtimes (`ConfigManager`, `load`,
 * `get`). Ver la fila D7 de PARITY.md.
 *
 * Requiere `npm run build` previo. Uso, desde la raíz del repo:
 *     node tests/fixtures/parity/emit.mjs
 */
import { readFileSync } from 'node:fs';
import { dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';

import { parse } from 'yaml';

const fixtureDir = dirname(fileURLToPath(import.meta.url));
const repoRoot = join(fixtureDir, '..', '..', '..');

const { ConfigManager } = await import(join(repoRoot, 'dist', 'index.js'));

process.chdir(fixtureDir);

const names = Object.keys(parse(readFileSync('config.yaml', 'utf8')).variables).sort();

const manager = new ConfigManager('config.yaml');
await manager.load();

const values = {};
for (const name of names) values[name] = manager.get(name);
console.log(JSON.stringify(values, null, 2));

File: tests/fixtures/parity/emit.py
"""Emisor de paridad — runtime Python.

Resuelve el fixture canónico e imprime en stdout el JSON de valores, ordenado
por clave y con indentación de 2. Nada más va a stdout.

Solo usa API pública común a ambos runtimes (`ConfigManager`, `load`, `get`):
así el emisor no depende de métodos que hoy existen en un runtime y no en el
otro. Ver la fila D7 de PARITY.md.

Uso, desde la raíz del repo:
    python tests/fixtures/parity/emit.py
"""

import json
import os
import sys
from pathlib import Path

import yaml

FIXTURE_DIR = Path(__file__).resolve().parent
REPO_ROOT = FIXTURE_DIR.parents[2]

sys.path.insert(0, str(REPO_ROOT / "src"))
os.chdir(FIXTURE_DIR)

from env_manager import ConfigManager  # noqa: E402

with open("config.yaml", encoding="utf-8") as handle:
    names = sorted(yaml.safe_load(handle)["variables"])

manager = ConfigManager("config.yaml")
manager.load()
print(json.dumps({name: manager.get(name) for name in names}, sort_keys=True, indent=2))
